File: exercism_data/src/582.rb
class Phrase
	attr_accessor :text

	def initialize(str)
		@text = str
	end
	
	def word_count
		result = Hash.new	
		@text.split(/\W/).reject(&:empty?).each do |word|
			word.downcase!
			result[word] = result[word].nil? ? 1 : result[word] + 1
		end
		result
	end
end

File: assignments/python/wc/src/582.py
def word_count(phrase):
    phrase = phrase.replace("\n"," ")
    words = phrase.split(" ")
    word_dict = {}
    try:
        while 1:
            words.remove('')
    except ValueError:
        pass
    for word in words:
        if word_dict.has_key(word):
            word_dict[word] = word_dict[word] + 1
        else:
            word_dict[word] = 1
    return word_dict
